crates/red_knot_python_semantic/src/types.rs

```diff
@@ -841,6 +854,15 @@ impl<'db> FunctionType<'db> {
             })
     }
 
+    /// Return true if this is a symbol with given name from `typing` or `typing_extensions`.
+    pub(crate) fn is_typing_symbol(self, db: &'db dyn Db, name: &str) -> bool {
+        name == self.name(db)
+            && file_to_module(db, self.definition(db).file(db)).is_some_and(|module| {
+                module.search_path().is_standard_library()
+                    && matches!(module.name(), "typing" | "typing_extensions")
```

Comment:
oops, this should fix the build failure introduced by my previous suggestion:

```suggestion
                    && matches!(&**module.name(), "typing" | "typing_extensions")
```

Reply:
or this:

```suggestion
                    && matches!(module.name().as_str(), "typing" | "typing_extensions")
```